src/features/expand-collapse-outdated-comments.js

```diff
@@ -0,0 +1,40 @@
+import select from 'select-dom';
+import {isMac, metaKey} from '../libs/utils';
+
+const keyName = isMac ? 'Alt' : 'Ctrl';
+
+function addTooltip(button) {
+	button.setAttribute('aria-label', keyName + ' + click to expand/collapse all outdated comments');
+	button.classList.add('tooltipped', 'tooltipped-n');
+}
+
+export default () => {
+	const showOutdatedButtons = select.all('.show-outdated-button, .hide-outdated-button');
+	showOutdatedButtons.forEach(button => {
+		addTooltip(button);
+		button.addEventListener('click', e => {
+			if (e[metaKey]) {
+				const parentElement = e.target.parentNode;
+				const viewportOffset = parentElement.getBoundingClientRect().top;
+				if (e.target.classList.contains('show-outdated-button')) {
+					select.all('.outdated-comment:not(.open) .show-outdated-button').forEach(button => {
+						if (button !== e.target) {
+							button.click();
+						}
+					});
+				} else {
+					select.all('.outdated-comment.open .hide-outdated-button').forEach(button => {
+						if (button !== e.target) {
+							button.click();
+						}
+					});
+				}
+				// Scroll to original position where the click occurred after the rendering of all click events is done
+				requestAnimationFrame(() => {
+					const offsetTop = $(parentElement).offset().top - viewportOffset;
+					window.scroll(0, offsetTop);
+				});
```

Comment:
Is this wait required or does it work without?

Reply:
I ask because even if “rendering takes some time” as you mentioned in a comment, the offset value should be updated in the same loop (relayout is synchronous)

Reply:
Hey, I made a small demo [here without rAF](https://gist.github.com/anonymous/ff824cce07b31030bfc757ae35919645), and [here with rAF](https://gist.github.com/anonymous/e1829724a41966ebd2e15e4d96cd9810). Both of them work the same as you described (relayout as synchronous).

But in case of GitHub (both in Chrome and Firefox), the scroll has to be done as a microtask, because GitHub has its own methods to run in the mean time (which are run as microtasks)

---------------------------
Without rAF:
![image](https://user-images.githubusercontent.com/12368291/33006109-167395e4-cdef-11e7-9133-dbcd6f8e1bf2.png)

---------------------------
With rAF:
![image](https://user-images.githubusercontent.com/12368291/33005996-8f4026dc-cdee-11e7-9f51-24f3bf376bd2.png)

---------------------------
GH's code that runs async (promise on line `8057`):
![image](https://user-images.githubusercontent.com/12368291/33006179-5ecb002a-cdef-11e7-869f-60010b6e2bc4.png)